### PR TITLE
feat: enforce space before asyncArrow function paren

### LINKED ISSUE #27: space-before-function-paren rules for 'asyncArrow'
I have faced the issue during reviewing the PR https://github.com/hexojs/hexo/pull/3954

Currently we are using `['error', 'never']`:

https://github.com/hexojs/eslint-config-hexo/blob/ada15707cc6a7b0e3331705005f0bfba692ec54a/eslint.js#L106

```js
function foo() {
    // ...
}

var bar = function() {
    // ...
};

var bar = function foo() {
    // ...
};

class Foo {
    constructor() {
        // ...
    }
}

var foo = {
    bar() {
        // ...
    }
};

var foo = async() => 1
```

IMHO we could add a space between `async` & `await` and function for better distinguishment (`['error', {"anonymous": "never", "named": "never", "asyncArrow": "always"}]`), which will resulted in:

```js
function foo() {
    // ...
}

var bar = function() {
    // ...
};

class Foo {
    constructor() {
        // ...
    }
}

var foo = {
    bar() {
        // ...
    }
};

var foo = async (a) => await a
```

> "anonymous" and "named" retain `never` for consistency.

### PR DESCRIPTION
**Pull request by SukkaW**
#27

### CODE REVIEW

File: eslint.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ module.exports = {
       after: true
     }],
     'space-before-blocks': 'error',
-    'space-before-function-paren': ['error', 'never'],
+    'space-before-function-paren': ['error', {'anonymous': 'never', 'named': 'never', 'asyncArrow': 'always'}],
```
Comment (curbengh) [inline eslint.js]: suggest to split the object into separate lines and remove quote on key.

example:
https://github.com/hexojs/eslint-config-hexo/blob/8ea0e70e8ec2dbf158d79fc9a03c80c0f72700fc/eslint.js#L94-L97

**Review by curbengh — CHANGES_REQUESTED** [review]
code style

File: eslint.js (hunk 1 above)
Comment (SukkaW) [reply eslint.js]: Updated.

**Review by curbengh — APPROVED** [review]
Tested on hexo repo, verified no breaking change.

**Pull request merged**